app/javascript/js/controllers/fields/easy_mde_controller.js

```diff
@@ -1,36 +1,106 @@
-import { Controller } from '@hotwired/stimulus'
-import EasyMDE from 'easymde'
+import { Controller } from "@hotwired/stimulus";
+import EasyMDE from "easymde";
+import { DirectUpload } from "@rails/activestorage";
 
 export default class extends Controller {
-  static targets = ['element']
+  static targets = ["element"];
 
   get view() {
-    return this.elementTarget.dataset.view
+    return this.elementTarget.dataset.view;
   }
 
   get componentOptions() {
     try {
-      return JSON.parse(this.elementTarget.dataset.componentOptions)
+      return JSON.parse(this.elementTarget.dataset.componentOptions);
     } catch (error) {
-      return {}
+      return {};
     }
   }
 
   connect() {
+    console.log("Hello from the connection")
+    
     const options = {
       element: this.elementTarget,
       spellChecker: this.componentOptions.spell_checker,
-      autoRefresh: { delay: 500},
+      autoRefresh: { delay: 500 },
+    };
+
+    if (this.view === "show") {
+      options.toolbar = false;
+      options.status = false;
     }
 
-    if (this.view === 'show') {
-      options.toolbar = false
-      options.status = false
+
+    if (this.componentOptions.image_upload) {
+      this._configureImageUploads(options);
     }
 
-    const easyMde = new EasyMDE(options)
-    if (this.view === 'show') {
-      easyMde.codemirror.options.readOnly = true
+    const easyMde = new EasyMDE(options);
+    if (this.view === "show") {
+      easyMde.codemirror.options.readOnly = true;
     }
   }
+
+  _configureImageUploads(options) {
```

Comment:
Let's use the JS private notiation please.
```suggestion
  #configureImageUploads(options) {
```

Then can be used liek so `this.#configureImageUploads()`